src/core/toga/app.py

```diff
@@ -75,6 +72,9 @@ def __init__(self, name, app_id,
 
         self._impl = self.factory.App(interface=self)
 
+        self._impl.document_types = document_types
```

Comment:
Are you sure this is right? The document types list should be on the interface, shouldn't it?

Reply:
You were right, it removed the error but didn't fix the underlying problem with the way the AppDelegate was trying to use the interface in the cocoa implementation.